File: mario-more/mario.c
// asks for height and then prints pyramid with that height
#include <cs50.h>
#include <stdio.h>

int main(void)
// print x spaces (where x = height - iteration) then print "#" iteration times then 2 spaces then "#" iteration times
{
    // asks input until input is in 1-8 range
    int height = get_int("Height: ");
    while (height < 1 || height > 8)
    {
        height = get_int("Height: ");
    }

    int iter = 1;
    do
    {
        int n = height - iter;
        int x;
        for (x = 0; x < n; x++)
        {
            printf(" ");
        }

        for (x = 0; x < iter; x++)
        {
            printf("#");
        }

        printf("  ");

        for (x = 0; x < iter; x++)
        {
            printf("#");
        }
        printf("\n");
        iter++;
    }
    while (iter <= height);
}

File: sentimental-mario-more/mario.py
"""
 Copyright (c) 2022, Ricardo Madeira
All rights reserved.

This source code is licensed under the BSD-style license found in the
LICENSE file in the root directory of this source tree.  
"""

# TODO
from cs50 import get_int


def main():
    # Asks input for height of pyramid until we get a value between 1-8
    height = get_int("Height: ")
    while (height < 1 or height > 8):
        height = get_int("Height: ")

    # Loop each row of the pyramid
    for i in range(1, height + 1):
        lead_spaces = height - i
        print(" " * lead_spaces, end="")
        print("#" * i, end="")
        print(" " * 2, end="")
        print("#" * i)


if __name__ == "__main__":
    main()
